### PR TITLE
Standardize usage of NVM

### PR DESCRIPTION
**Pull request by ObliviousHarmony**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

As we converge on the creation of a monorepo it becomes increasingly necessary that we are all using the same version of Node and NPM when working on WooCommerce Core. When working on various PRs I've encountered issues caused by `package-lock.json` files in different packages having been generated with different versions of NPM. It is for this reason that I think we should adopt the use of [NVM](https://github.com/nvm-sh/nvm) in order to standardize our Node versions.

This PR adds a `.nvmrc` file which can be consumed by the `nvm use` and `nvm install` commands. ~~I've also adjusted Travis so that it will run the CI install command~~(See commits for explanation) as well as install the version of Node described in the `.nvmrc` file.

### How to test the changes in this Pull Request:

1. [Install NVM (if you don't already have it)](https://github.com/nvm-sh/nvm)
2. `nvm install` in the root WC directory
3. `rm -rf node_modules tests/e2e/env/node_modules tests/e2e/factories/node_modules tests/e2e/utils/node_modules`
4. `npm install` should quickly install dependencies for WooCommerce Core and then all of the packages.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### CODE REVIEW

**Comment by rrennick** [comment]
@ObliviousHarmony Just checking- I edited the testing instructions to change `npm run install` to `npm install` as `run install` gave me an error.

**Review by rrennick — APPROVED** [review]
@ObliviousHarmony This worked great locally with the changes to the testing instructions.

**Review by vedanshujain — COMMENTED** [review]
Works great for me, I but I have not dived deep into reviewing and will leave the approving to folks more knowledgeable about this though.

**Review by rodrigoprimo — COMMENTED** [review]
I can confirm that simply removing `woocommerce` from `tests/e2e/env/package.json` fixes the problem that I was having when running `npm install`. It is not clear to me what is the relation between standardizing the usage of NVM and the fix to the problem that I reported in the Slack channel. Maybe this PR should be split in two? Or could you please clarify what is the relation between `woocommerce` from `tests/e2e/env/package.json` and standardizing the usage of NVM?

**Comment by rrennick** [comment]
> Or could you please clarify what is the relation between woocommerce from tests/e2e/env/package.json

@rodrigoprimo That was a leftover from when the original POC PR was in the `woocommerce-admin` repo. We discovered the issue with node/npm versions due to that being missed when the package code was ported to this repo.

File: .nvmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+v10
```
Comment (jonathansadowski) [inline .nvmrc:1]: Should this be `v10.22` to match package.json?

Comment (jonathansadowski) [reply .nvmrc:1]: Scratch that, I forgot what `^` meant :)

**Comment by andfinally** [comment]
Install worked nicely for me too! 🥳 There were a few warnings at the end about some packages needing peer dependencies, but I assume that's not unusual for an npm install.

**Review by claudiosanches — APPROVED** [review]

**Review by jonathansadowski — APPROVED** [review]
Works well. `nvm` ❤️

**Pull request merged**